x-pack/legacy/plugins/siem/public/components/events_viewer/events_viewer.tsx

```diff
@@ -139,73 +145,81 @@ export const EventsViewer = React.memo<Props>(
                     pageInfo,
                     refetch,
                     totalCount = 0,
-                  }) => (
-                    <>
-                      <HeaderSection
-                        id={id}
-                        showInspect={showInspect}
-                        subtitle={
-                          utilityBar
-                            ? undefined
-                            : `${i18n.SHOWING}: ${totalCount.toLocaleString()} ${i18n.UNIT(
-                                totalCount
-                              )}`
-                        }
-                        title={timelineTypeContext?.title ?? i18n.EVENTS}
-                      >
-                        {headerFilterGroup}
-                      </HeaderSection>
+                  }) => {
+                    const totalCountMinusDeleted =
+                      totalCount > 0 ? totalCount - deletedEventIds.length : 0;
+
+                    // TODO: Reset eventDeletedIds/eventLoadingIds on refresh/loadmore (getUpdatedAt)
```

Comment:
Maybe TimelineQuery can take care of that if you passed the parameter and default it.

Reply:
Perfect! I was hesitant to move that logic into `TimelineQuery` and wanted to chat with you first -- this will work out well.